mcs/class/System.Private.CoreLib/System/AppDomain.cs

```diff
@@ -0,0 +1,28 @@
+using System.Reflection;
+using System.Runtime.InteropServices;
+
+namespace System {
+
+	[StructLayout (LayoutKind.Sequential)]
+	public sealed partial class AppDomain : MarshalByRefObject {
```

Comment:
It definitely cannot be public as it'd conflict with CoreFX System.AppDomain. Could we kill this or at least rename this to something else?

Reply:
killing it will be a big project.  I'll see if we can give it a different name for netcore builds...

Reply:
We can name it 'MonoDomain' or 'RuntimeDomain'.

Reply:
or move it do different namespace, please make it internal